optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/OptaShiftRosteringEntryPoint.java

```diff
@@ -35,7 +42,27 @@
 
     @PostConstruct
     public void onModuleLoad() {
-        tenantStore.init(); //FIXME: Shouldn't this call be made by the Container once it's annotated with @PostConstruct?
+        GWT.setUncaughtExceptionHandler(new GWT.UncaughtExceptionHandler() {
+
+            public void onUncaughtException(Throwable e) {
+                Throwable unwrapped = unwrap(e);
+                StringBuilder message = new StringBuilder();
+                StackTracePrintStream stackTracePrintStream = new StackTracePrintStream(message);
+                unwrapped.printStackTrace(stackTracePrintStream);
+                ErrorPopup.show(message.toString());
+            }
+
+            public Throwable unwrap(Throwable e) {
+                if (e instanceof UmbrellaException) {
+                    UmbrellaException ue = (UmbrellaException) e;
+                    if (ue.getCauses().size() == 1) {
+                        return unwrap(ue.getCauses().iterator().next());
+                    }
+                }
+                return e;
+            }
+        });
```

Comment:
Not a blocker, but I think this code block could use a method extraction.

Reply:
Extract what? There is only one statement in `onModuleLoad`. The only thing we can do is move the implementation to another class.